### PR TITLE
feat(BMS): bms support edit metadata

### PR DESCRIPTION
**Pull request by wanglong3710**
**What this PR does / why we need it**:
bms support edit metadata

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
To avoid triggering changes metadata is not backfilled during read.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/bms/' TESTARGS='-run TestAccBmsInstance_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms/ -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4 
=== RUN   TestAccBmsInstance_basic 
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (543.70s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       543.740s

```

### CODE REVIEW

File: huaweicloud/services/bms/resource_huaweicloud_bms_instance.go [hunk 1]
```diff
@@ -440,6 +457,22 @@ func resourceBmsInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
+	if d.HasChange("agency_name") {
+		metadataOpts := make(map[string]interface{})
+		metadataOpts["agency_name"] = d.Get("agency_name").(string)
+		_, err := baremetalservers.UpdateMetadata(bmsClient, d.Id(), metadataOpts).Extract()
+		if err != nil {
+			return diag.Errorf("error updating the bms metadata agency_name: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/bms/resource_huaweicloud_bms_instance.go]:
```suggestion
			return diag.Errorf("error updating the BMS metadata agency_name: %s", err)
```
Please check the other parts.

File: huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go [hunk 2]
```diff
@@ -209,6 +221,15 @@ resource "huaweicloud_bms_instance" "test" {
     tag2 = "value2"
   }
 
+  system_disk_type = "GPSSD"
+  system_disk_size = 150
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go]: Is it necessary to add `system_disk_type` and `system_disk_size` when testing `metadata`?

File: huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go [hunk 3]
```diff
@@ -163,10 +167,18 @@ resource "huaweicloud_bms_instance" "test" {
...
+  agency_name = "aom_admin_trust"
+
+  metadata = {
+    __bms_support_evs = "true"
+  }
+
   charging_mode = "prePaid"
   period_unit   = "month"
   period        = "1"
-  auto_renew    = "false"
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go]: Please explain the reason why removing `auto_renew`.
This kind of change needs to be explained in the PR.

File: huaweicloud/services/bms/resource_huaweicloud_bms_instance.go [hunk 4]
```diff
@@ -440,6 +457,22 @@ func resourceBmsInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
+	if d.HasChange("agency_name") {
+		metadataOpts := make(map[string]interface{})
+		metadataOpts["agency_name"] = d.Get("agency_name").(string)
```
Comment (deer-hang) [inline huaweicloud/services/bms/resource_huaweicloud_bms_instance.go]:
```
	opts := map[string]interface{}{
		"agency_name": d.Get("agency_name").(string),
	}
```
It is more intuitive to write map like this.

File: huaweicloud/services/bms/resource_huaweicloud_bms_instance.go [hunk 5]
```diff
@@ -240,7 +240,12 @@ func ResourceBmsInstance() *schema.Resource {
 			"agency_name": {
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
+				Computed: true,
+			},
+			"metadata": {
```
Comment (deer-hang) [inline huaweicloud/services/bms/resource_huaweicloud_bms_instance.go:244]: Please explain why metadata is not backfilled during read.
I think it's better to add a comment.

File: docs/resources/bms_instance.md [hunk 6]
```diff
@@ -189,8 +189,14 @@ The following arguments are supported:
 * `auto_renew` - (Optional, String) Specifies whether auto renew is enabled. Valid values are "true" and "
   false", defaults to *false*.
 
-* `agency_name` - (Optional, String, ForceNew) Specifies the IAM agency name which is created on IAM to provide
-  temporary credentials for BMS to access cloud services. Changing this creates a new instance.
+* `agency_name` - (Optional, String) Specifies the IAM agency name which is created on IAM to provide
+  temporary credentials for BMS to access cloud services. User can define the value.
```
Comment (ShiChangkuo) [inline docs/resources/bms_instance.md]: what does `User can define the value` mean?

File: docs/resources/bms_instance.md [hunk 7]
```diff
@@ -189,8 +189,14 @@ The following arguments are supported:
...
+
+* `metadata` - (Optional, Map) Specifies the user-defined metadata key-value pair. If you do not specify any
+  key-value pair, metadata will not be updated.
```
Comment (ShiChangkuo) [inline docs/resources/bms_instance.md]: why should we add `If you do not specify any key-value pair, metadata will not be updated` ?

File: docs/resources/bms_instance.md [hunk 8]
```diff
@@ -189,8 +189,14 @@ The following arguments are supported:
...
+  + A metadata key contains of a maximum of 255 Unicode characters which can be letters, digits, hyphens (-),
+    underscores (_), colons (:), and point (.). Key cannot be left blank.
```
Comment (ShiChangkuo) [inline docs/resources/bms_instance.md]: I think we can delete ` Key cannot be left blank.`

**Pull request merged**